torch/_inductor/codegen/wrapper.py

```diff
@@ -1958,28 +1972,97 @@ def codegen_unbacked_symbol_decl(self, symbol):
+        # codegen infra for cpp_wrapper=True case. Therefore for cpp_wrapper, we
+        # fallback to the old way of inlining subgraphs in the output code. Once
+        # we update CppWrapperCpu, we can remove this function.
+        def _codegen_subgraph_prefix():
+            for inner_input, outer_input in zip(
+                subgraph.graph.graph_inputs, outer_inputs
+            ):
+                self.writeline(
+                    f"{self.declare}{inner_input} = {outer_input}{self.ending}"
+                )
+
+        def _codegen_subgraph_suffix():
+            for inner_output, outer_output in zip(
+                subgraph.graph.graph_outputs, outer_outputs
+            ):
+                self.writeline(
+                    f"{outer_output} = {inner_output.codegen_reference()}{self.ending}"
+                )
+
+        try:
+            self.push_codegened_graph(subgraph.graph)
+            self.writeline(f"{self.comment} subgraph: {subgraph.name}")
+            _codegen_subgraph_prefix()
+            parent_graph = V.graph
+            with V.set_graph_handler(subgraph.graph):
+                subgraph.graph.codegen_subgraph(
+                    parent_graph=parent_graph,
+                )
+            _codegen_subgraph_suffix()
+        finally:
+            self.pop_codegened_graph()
+
     def codegen_subgraph_prefix(self, subgraph, outer_inputs, outer_outputs):
+        subgraph.graph.add_symbol_graph_inputs()
         for inner_input, outer_input in zip(subgraph.graph.graph_inputs, outer_inputs):
             self.writeline(f"{self.declare}{inner_input} = {outer_input}{self.ending}")
 
     def codegen_subgraph_suffix(self, subgraph, outer_inputs, outer_outputs):
         for inner_output, outer_output in zip(
-            subgraph.graph.graph_outputs, outer_outputs
+            subgraph.graph.get_output_names(), outer_outputs
         ):
-            self.writeline(
-                f"{outer_output} = {inner_output.codegen_reference()}{self.ending}"
-            )
+            self.writeline(f"{outer_output} = {inner_output}{self.ending}")
+
+    def codegen_subgraph_call(self, subgraph, outer_inputs, outer_outputs):
+        # Get the input and output names of the subgraph
+        inner_inputs = ", ".join(subgraph.graph.graph_input_names)
+        if len(subgraph.graph.graph_input_names) == 1:
+            inner_inputs += ","
+
+        output_names = subgraph.graph.get_output_names()
+        inner_outputs = ", ".join(output_names)
+        if len(output_names) == 1:
+            inner_outputs += ","
+
+        # Call the subgraph launcher function
+        self.writeline(f"({inner_outputs}) = {subgraph.graph.name}([{inner_inputs}])")
 
     def codegen_subgraph(self, subgraph, outer_inputs, outer_outputs):
+        # Codegen subgraph by recursively calling the codegen for the subgraph.
+        # This lifts the subgraph as a function in the output code.
+        if V.graph.aot_mode:
```

Comment:
So this has to be `aot_mode`. The reason is that for aot_mode, we call two passes (one with CPP and one withoug CPP wrapper). And there is information sharing between the two passes (like names of the generated kernels). So, these two have to follow same subgraph codegen mechanism. So, this falls in the same bucket as support lifting for cpp_wrapper.

Reply:
I see, but cpp_wrapper+JITInductor also does two passes codegen for GPU. So I think instead of checking V.graph.cpp_wrapper, we should check config.cpp_wrapper.

Reply:
This still has to be aot_mode. Many tests fail if I switch to `V.graph.cpp_wrapper`.

`pytest test/inductor/test_aot_inductor.py::AOTInductorTestNonABICompatibleCuda::test_cond_with_reinterpret_view_inputs_outputs_non_abi_compatible_cuda`